src/wp-includes/class-wp-scripts.php

```diff
@@ -133,6 +133,24 @@ class WP_Scripts extends WP_Dependencies {
 	 */
 	private $type_attr = '';
 
+	/**
+	 * Holds a mapping of dependents (as handles) for a given script handle.
+	 * Used to optimize recursive dependency tree checks.
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	private $dependents_map = array();
+
+	/**
+	 * Holds a reference to the allowed script loading strategies.
+	 * Used by methods that validate loading strategies.
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	public $allowed_strategies = array( 'blocking', 'defer', 'async' );
```

Comment:
I think this should rather be `private`. If we want it to be extensible, we should look at that later.

```suggestion
	private $allowed_strategies = array( 'blocking', 'defer', 'async' );
```

Alternatively, as with PHP 5.6, this could instead be a `const` like:

```php
	const ALLOWED_STRATEGIES = array( 'blocking', 'defer', 'async' );
```

But this wouldn't be able to be a `private` const before PHP 7.1.

Reply:
@westonruter fair suggestion thank you, I've opted for the private property as opposed to the const approach (for the access restriction alone). I'm in agreement that if strategies need to be extended in future, we can look at various means thereof.

I've also updated the return type to be `string[]` as opposed to `array`, something that hopefully starts to become second nature to me :)

The commit containing these fixes can be seen at https://github.com/WordPress/wordpress-develop/pull/4391/commits/d17203e911fccaed4d584feb4e0f8759f3fae590